crates/iceberg-catalog/src/catalog/views/load.rs

```diff
@@ -74,7 +74,8 @@ pub(crate) async fn load_view<C: Catalog, A: Authorizer + Clone, S: SecretStore>
 
     let view_location = parse_view_location(&view_metadata.location)?;
 
-    // We don't commit the transaction yet, first we need to write the metadata file.
+    t.commit().await?;
```

Comment:
If we commit here, the rest of the function must be infallible. The commit succeeded so we must return an ok value to the client. Lets also double check tables. Alternatively we can get the secret earlier?

Reply:
This is a load fn, a read only transaction, committing it should have no sideffects but dropping the connection

Reply:
Ah - you are right! Should have used a proper screen.